### PR TITLE
Reject invalid rational literals (e.g. `1e1r`) on lexing

### LINKED ISSUE #1586: YARP accepts `1e1r` and `1e1ri` but Ruby rejects it
```ruby
eval('1e1r')
# => syntax error, unexpected local variable or method, expecting end-of-input (SyntaxError)

eval('1e1ri')
# => syntax error, unexpected local variable or method, expecting end-of-input (SyntaxError)

YARP.parse('1e1r').success?
# => true
YARP.parse('1e1ri').success?
# => true
```

Maybe it should be accepted in Ruby because Ruby accepts `1.0ri` and `1e1i`

### PR DESCRIPTION
**Pull request by makenowjust**
Fix #1586

### CODE REVIEW

File: src/prism.c [hunk 1]
```diff
@@ -6452,7 +6454,8 @@ lex_numeric(pm_parser_t *parser) {
     parser->integer_base = PM_INTEGER_BASE_FLAGS_DECIMAL;
 
     if (parser->current.end < parser->end) {
-        type = lex_numeric_prefix(parser);
+        bool seen_e = false;
```
Comment (makenowjust) [inline src/prism.c:6457]: The variable name `seen_e` is derived from CRuby's implementation. If anyone has a better name, please suggest it to me.

**Review by kddnewton — APPROVED** [review]
Nice! `seen_e` is fine, I think that's clear.

**Comment by makenowjust** [comment]
Thank you.

@kddnewton A test for invalid cases is missing in this PR now. Could you tell me how to test invalid cases?

**Comment by kddnewton** [comment]
@makenowjust we typically put those in `test/prism/errors_test.rb`.

**Pull request merged**